CustomCollections/CustomCollections/CustomVector.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomVector<T> : IList<T>
+    {
+        private T[] _vector;
+
+        public int Count { get; private set; }
+        public bool IsReadOnly { get; private set; } = true;
+
+        public CustomVector()
+        {
+            _vector = new T[0];
+            Count = _vector.Length;
+        }
+
+        public T this[int index]
+        {
+            get => _vector[index];
+            set => _vector[index] = value;
+        }
+
+        public void Add(T item)
+        {
+            var newVector = new T[Count + 1];
+            for (int i = 0; i < Count; i++)
+            {
+                newVector[i] = _vector[i];
+            }
+            newVector[Count] = item;
+            Count++;
+            _vector = newVector;
+            IsReadOnly = false;
+        }
+
+        public void Clear()
+        {
+            _vector = new T[0];
+            Count = 0;
+            IsReadOnly = true;
+        }
+
+        public bool Contains(T item)
+        {
+            if (_vector.Contains(item))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public bool Remove(T item)
+        {
+            _vector = _vector.Where(s => !s.Equals(item)).ToArray();
+            if (Count > _vector.Length)
+            {
+                Count = _vector.Length;
+                return true;
+            }
+            return false;
+        }
+
+        public void RemoveAt(int index)
+        {
+            Remove(_vector[index]);
```

Comment:
This is bad implementation. If you have vector `(1 2 3 4 5 3 6)` and call `RemoveAt(5)`, your implementation of `Remove` will remove items at indexes 2 and 5, while correct implementation will only remove item at 2. Both are not what user expects. But even bigger problem here is ineffectiveness of this implementation. Your general implementation of `Remove(item)` without Linq would look like `IndexOf(item)` and then `if (index > -1) RemoveAt(index)`. No need to search for item in the collection, you already have it's index!